docs/how_to/add_doc_index.md

```diff
@@ -402,6 +402,16 @@ in your backend, if such a concept exists in your case. In your implementation y
 (Strictly speaking, this uniqueness property is not guaranteed, since a user could override the auto-generated `.id` field with a custom value.
 If your implementation encounters a duplicate `.id`, it is okay to fail and raise an Exception.)
 
+### The `_filter_by_parent_id()` method
+
+The default implementatin return `None`. You can choose to override this function with database specific filter API when needed. 
+This function should return a list of ids of subindex level documents given the id of root document.
+
+### The `index_name()` property
+
+The `index_name` property is used in the initialization of subindices, and the default implementation is empty. This function should return the name of the index. And if the property of the index name in your backend is not `index_name`, you need to convert it as the first step in `__init__()`, like `index_name` is assigned to `work_dir` in `docarray/index/backends/hnswlib.py`.
```

Comment:
For non-subindex the default behaviour is now to use the class name of the schema, see here: https://github.com/docarray/docarray/pull/1486

I think for subindex something similar should apply:

```python
class Doc(BaseDoc):
    inner = DocList[Bla]
```
I think the subindex for `inner` here should have a reasonable default index name.
Maybe `inner_Bla` to avoid conflict with other indices of the `Bla` class?

@samsja @JoanFM I'm als curious about your input here

Reply:
it would not be a default however, it should be defined by thr parent indrx right? I think the name of the subindex should be fully determined by the name og the root and the class and name of the subindex

Reply:
> it would not be a default however, it should be defined by thr parent indrx right? I think the name of the subindex should be fully determined by the name og the root and the class and name of the subindex

Yes that would be nice, however there are differences in what the different backends accept as a valid name. Therefore just doing `rootname + subname` or `rootname + "_" + subname` may not work for all of them; but great if it does!

Reply:
`index_name` property is used when getting the `rootname`:
```python
sub_db_config.index_name = f'{self.index_name}__{col_name}'
```